### ISSUE TITLE
Add support for Pytorch Lightning Trainer.test

### ISSUE THREAD

**Issue by AlexandreBrown** [airctic/icevision#1124]
## 🚀 Feature
**Is your feature request related to a problem? Please describe.**
Currently, it is not possible to test an IceVision model with Pytorch Lightning (we can only train it but cannot test it). 

**Describe the solution you'd like**
We should override the `test_step` method in the IceVision Pytorch Lightning Model Adapter class so that it can be used by the Pytorch Lightning Trainer class (see https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#testing ) 

**Describe alternatives you've considered**
I've considered using the `trainer.validate` since IceVision already overrides the `validation_step` but this is not ideal since the validation step logs using [`self.log("val_loss", loss)`](https://github.com/airctic/icevision/blob/356d29ba3ddc0d51c88e6c4ed78ac9bfb4dde4cc/icevision/models/ultralytics/yolov5/lightning/model_adapter.py#L58) which means that test metrics would be grouped with validaton metrics (eg: Using wandb logging).   
This is not ideal since we lose the ability to clearly distinct validation metrics from test metrics.

**Additional context**
PL Error : 
```
MisconfigurationException: No `test_step()` method defined to run `Trainer.test`.
```

**Comment by AlexandreBrown** [comment]
I will start to work on this tomorrow.  
I see a great opportunity to group some common code that seems to be present in all the `LightningModuleAdapter` implementations.
